### ISSUE TITLE
Convert `root()` to `all(Class)` selector when using `setModel()` with collections

### ISSUE THREAD

**Issue by mmefremov** [instancio/instancio#1215]
**Describe the bug**
#1160 reproduced with collections.

**Expected result**
The collection size settings are set within the current model.

**Actual result**
The collection size for the current model sets the size for all collection models.

**Additional information**
accountNumber should have size 1, principalSchedule should have according to the default setting for collections (from 2 to 6).

**Examples**
I set maxSize in the method _createGqlAccountNumberListModel()_ according to the _accountNumber_ size (it's equal 1)
```java
    public static Model<GqlCreateLoanAgreementInput> createGqlCreateLoanAgreementInputModel(List<CamAccountCardEntity> accountCardList) {
        return Instancio.of(GqlCreateLoanAgreementInput.class)
                .setModel(field(GqlCreateLoanAgreementDetails::getAccountNumbers), createGqlAccountNumberListModel(accountCardList))
                .setModel(field(GqlCreateLoanAgreementInput::getInterestSchedule), createGqlInterestRepaymentDueDayModel())
                .toModel();
 
    private static Model<List<GqlAccountNumber>> createGqlAccountNumberListModel(List<CamAccountCardEntity> accountCardList) {
        return Instancio.ofList(GqlAccountNumber.class)
                .generate(root(), gen -> gen.collection().maxSize(accountCardList.size()))
                .generate(field(GqlAccountNumber::getOrder), Generators::intSeq)
                .supply(field(GqlAccountNumber::getAccountNumber), () -> accountCardList.listIterator().next().getAccountNumber())
                .toModel();
    }
 
    private static Model<List<GqlDueInput>> createPrincipalScheduleListModel() {
        return Instancio.ofList(GqlDueInput.class)
                .setBlank(field(GqlDueInput::getDueAmount))
                .set(field(GqlDueInput::getDueDate), LocalDate.now())
                .toModel();
    }
```
After the object is created, both collections have a size of 1:
```
[INFO  ] input.accountNumbers: [GqlAccountNumber {accountNumber: 4000000000413INR0014, order: 1}]
[INFO  ] input.principalSchedule: [GqlDueInput {dueActualAmount: null, dueAmount: 97600, dueDate: 2093-01-23}]
```
Now I add _minSize(2)_ for the method _createPrincipalScheduleListModel()_
```java
    private static Model<List<GqlDueInput>> createPrincipalScheduleListModel() {
        return Instancio.ofList(GqlDueInput.class)
                .generate(root(), gen -> gen.collection().minSize(2))
                .setBlank(field(GqlDueInput::getDueAmount))
                .set(field(GqlDueInput::getDueDate), LocalDate.now())
                .toModel();
    }
```
Now the size settings in the method _createGqlAccountNumberListModel_ are ignored and the collection will also have a minimum size of 2
```
[INFO  ]input.accountNumbers: [GqlAccountNumber {accountNumber: 4000000000315INR0019, order: 1}, GqlAccountNumber {accountNumber: 4000000000315INR0019, order: 2}, GqlAccountNumber {accountNumber: 4000000000315INR0019, order: 3}, GqlAccountNumber {accountNumber: 4000000000315INR0019, order: 4}]
[INFO  ]input.principalSchedule: [GqlDueInput {dueAmount: 19700, dueDate: 2098-01-07}, GqlDueInput {dueAmount: 19700, dueDate: 2098-02-07}, GqlDueInput {dueAmount: 19700, dueDate: 2098-03-07}, GqlDueInput {dueAmount: 19700, dueDate: 2098-04-07}, GqlDueInput {dueAmount: 19700, dueDate: 2098-05-07}]
```

- Java version: 17
- Instancio version: 5.0.2

**Comment by armandino** [comment]
@mmefremov thanks for raising this issue! There's a lot going on in the example which makes it tricky to figure out what's going on. Any chance you could create a sample project so I can run the code? Simplifying it to the absolute minimum would help a lot.

**Comment by mmefremov** [comment]
Thanks for the quick reply. I'll try to have it done by noon.

**Comment by armandino** [comment]
@mmefremov thank you! I will look into it.

**Comment by mmefremov** [comment]
> Btw I believe this line should be size(numbers.size()) instead of maxSize(numbers.size()) for the assertion to pass.
You're right
> Could you please try 5.1.0-SNAPSHOT to verify?
I checked and everything works. Thank you very much for your work! Looking forward to the new version in the maven repository ))

**Comment by armandino** [comment]
@mmefremov glad to hear it! I just released `5.1.0`. It should be available from Maven central shortly. Let me know if you run into any other issues.
